eos/c-decays/dq-to-l-nu.cc

```
@@ -0,0 +1,180 @@
+
+        UsedParameter g_fermi;
+
+        UsedParameter m_Dq;
+
+        UsedParameter f_Dq;
+
+        UsedParameter tau_Dq;
+
+        LeptonFlavorOption opt_l;
+
+        UsedParameter m_l;
+
+        SpecifiedOption opt_cp_conjugate;
+
+        bool cp_conjugate;
+
+        UsedParameter mu;
+
+        std::function<double (const double &)> m_D_msbar;
+        std::function<complex<double> ()> v_cD;
+        std::function<WilsonCoefficients<ChargedCurrent> (LeptonFlavor, bool)> wc;
+
+        static const std::vector<OptionSpecification> options;
+
+        Implementation(const Parameters & p, const Options & o, ParameterUser & u) :
+            opt_model(o, options, "model"),
+            model(Model::make(opt_model.value(), p, o)),
+            opt_q(o, options, "q"),
+            hbar(p["QM::hbar"], u),
+            g_fermi(p["WET::G_Fermi"], u),
+            m_Dq(p["mass::D_" + opt_q.str()], u),
+            f_Dq(p["decay-constant::D_" + opt_q.str()], u),
+            tau_Dq(p["life_time::D_" + opt_q.str()], u),
+            opt_l(o, options, "l"),
+            m_l(p["mass::" + opt_l.str()], u),
+            opt_cp_conjugate(o, options, "cp-conjugate"),
+            cp_conjugate(destringify<bool>(opt_cp_conjugate.value())),
+            mu(p[opt_q.str() + "c" + "nu" + opt_l.str() + opt_l.str() + "::mu"], u)
+        {
+            Context ctx("When constructing D_q^+->l^+nu observable");
+
+            switch (opt_q.value())
+            {
+                case QuarkFlavor::strange:
+                    m_D_msbar = [this](const double & mu) -> double { return model->m_s_msbar(mu); };
+                    v_cD      = [this]() -> complex<double> { return model->ckm_cs(); };
+                    wc        = [this](LeptonFlavor l, bool cp) -> WilsonCoefficients<ChargedCurrent> { return model->wet_scnul(l, cp); };
+                    break;
+                default:
+                    throw InternalError("Invalid quark flavor: " + stringify(opt_q.value()));
```

Do we want to throw a specific error saying "not implemented" for the down case? As is done for the D_q -> P l nu

---

No in this case, since we disallow ``q=d`` through the options already.